lib/_http_agent.js

```diff
@@ -176,6 +180,7 @@ Agent.prototype.addRequest = function addRequest(req, options) {
     // If we are under maxSockets create a new one.
     this.createSocket(req, options, function(err, newSocket) {
       if (err) {
+        setInitTriggerId(newSocket._handle.getAsyncId());
```

Comment:
I generalized the call b/c of the not-so-obvious way that constructors like `TCPConnectWrap()` works. `setInitTriggerId()` is a little devil child that I'd love to banish to the depths of hell, but every time I've attempted to remove it there are complications.

Anyhow, I had considered creating an internal `nextTick()` that accepts the `triggerId` but didn't know how people would feel about it. I'll be happy to add it in if others are okay with it being introduced.